### ISSUE TITLE
SNOW-1822242: AttributeError: module 'snowflake.sqlalchemy.snowdialect' has no attribute 'BOOLEAN'

### ISSUE THREAD

**Issue by NathanFarmer** [snowflakedb/snowflake-sqlalchemy#549]
1. What version of Python are you using?

> 3.12.7

2. What operating system and processor architecture are you using?

> macOS-14.7.1-arm64-arm-64bit

3. What are the component versions in the environment (`pip freeze`)?

> asn1crypto==1.5.1
> certifi==2024.8.30
> cffi==1.17.1
> charset-normalizer==3.4.0
> cryptography==43.0.3
> filelock==3.16.1
> idna==3.10
> packaging==24.2
> platformdirs==4.3.6
> pycparser==2.22
> PyJWT==2.10.0
> pyOpenSSL==24.2.1
> pytz==2024.2
> requests==2.32.3
> snowflake-connector-python==3.12.3
> snowflake-sqlalchemy==1.7.0
> sortedcontainers==2.4.0
> SQLAlchemy==2.0.36
> tomlkit==0.13.2
> typing_extensions==4.12.2
> urllib3==2.2.3

4. What did you do?

```zsh
>> python3.12 -m venv snowflake-sqlalchemy && source snowflake-sqlalchemy/bin/activate
>> pip install snowflake-sqlalchemy==1.7.0
>> python
>> from snowflake.sqlalchemy import snowdialect
>> snowdialect.BOOLEAN
```

```zsh
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'snowflake.sqlalchemy.snowdialect' has no attribute 'BOOLEAN'
```

5. What did you expect to see?

```zsh
>> python3.12 -m venv snowflake-sqlalchemy && source snowflake-sqlalchemy/bin/activate
>> pip install 'snowflake-sqlalchemy<1.7.0'
>> python
>> from snowflake.sqlalchemy import snowdialect
>> snowdialect.BOOLEAN
```

```zsh
<class 'sqlalchemy.sql.sqltypes.BOOLEAN'>
```

**Comment by sfc-gh-dszmolka** [comment]
we'll check this, thank you for reporting!

**Comment by sfc-gh-dszmolka** [comment]
fix incoming on https://github.com/snowflakedb/snowflake-sqlalchemy/pull/551

**Comment by sfc-gh-dszmolka** [comment]
change is merged, but reopening this issue to be able to properly track the release

**Comment by sfc-gh-dszmolka** [comment]
released with snowflake-sqlalchemy v1.7.1
